agentstack/cli/tools.py

```diff
@@ -1,31 +1,44 @@
 from typing import Optional
 import itertools
+import sys
 import inquirer
 from agentstack.utils import term_color
 from agentstack import generation
 from agentstack.tools import get_all_tools
 from agentstack.agents import get_all_agents
+from agentstack.exceptions import ToolError, ValidationError
 
 
 def list_tools():
     """
     List all available tools by category.
     """
-    tools = get_all_tools()
-    curr_category = None
+    try:
+        tools = get_all_tools()
+        curr_category = None
 
-    print("\n\nAvailable AgentStack Tools:")
-    for category, tools in itertools.groupby(tools, lambda x: x.category):
-        if curr_category != category:
-            print(f"\n{category}:")
-            curr_category = category
-        for tool in tools:
-            print("  - ", end='')
-            print(term_color(f"{tool.name}", 'blue'), end='')
-            print(f": {tool.url if tool.url else 'AgentStack default tool'}")
+        print("\n\nAvailable AgentStack Tools:")
+        for category, tools in itertools.groupby(tools, lambda x: x.category):
+            if curr_category != category:
+                print(f"\n{category}:")
+                curr_category = category
+            for tool in tools:
+                print("  - ", end='')
+                print(term_color(f"{tool.name}", 'blue'), end='')
+                print(f": {tool.url if tool.url else 'AgentStack default tool'}")
 
-    print("\n\n✨ Add a tool with: agentstack tools add <tool_name>")
-    print("   https://docs.agentstack.sh/tools/core")
+        print("\n\n✨ Add a tool with: agentstack tools add <tool_name>")
+        print("   https://docs.agentstack.sh/tools/core")
+    except ToolError:
+        print(term_color("Could not retrieve list of tools. The tools directory may be corrupted or missing.", 'red'))
+        sys.exit(1)
+    except ValidationError:
```

Comment:
in only the case of a validation error, it'd be smart to log the exception because it will help them track down what's failing in the validation of their tool config.